styleguide/javascript/README.md

```diff
@@ -0,0 +1,147 @@
+# JavaScript Styleguide
+
+## Tooling
+
+### Avoid jQuery
+
+Just don't use it. It's better to use standard APIs.
+
+[Should you use or learn jQuery in 2020?](https://flaviocopes.com/jquery/)
+
+### Avoid lodash
+
+You probably don't need it. And if you do, import the functions you need like this:
+
+``` javascript
+import foo from 'lodash/foo'
+```
+
+Don't do this:
+
+``` javascript
+import { foo } from 'lodash'
+import _ from 'lodash'
+```
+
+The latter will make tree-shaking impossible.
+Read more about [lodash and tree-shaking](https://www.azavea.com/blog/2019/03/07/lessons-on-tree-shaking-lodash/)
+
+### Use Prettier
+
+Use our [prettier-config](https://github.com/barsoom/prettier-config)
+
+### Use eslint
+
+Use [eslint-config-barsoom](https://github.com/barsoom/eslint-config-barsoom)
+
+### When using Babel
+
+Try to avoid using too many experimental features.
+
+Stage 3 is fine and stage 2 can be OK.
+
+Just don't go crazy with everything shiny and new that is likely to change, or maybe not even get into the standard.
+
+### Prefer npm over yarn, unless you need transitive dependency pinning
```

Comment:
Does Rails still default to yarn? If so, is using npm instead of yarn with Rails straightforward or does it come with "fighting against the conventions" pains?

Reply:
Short answer: Rails defaults to yarn.

My take: For non-Rails apps, we could be more `npm`?

Reply:
I'd say, don't change anything, like, if Rails uses yarn, then we use yarn in that project.

If we start a new project, use npm, if we need some feature that exists in yarn but not in npm, use yarn.

To me it just makes more sense to default to using `npm` instead of `yarn` since you have to use npm to install yarn anyways, and they solve the same problem.

Reply:
Right. Could we make that explicit, then?

Reply:
Removed the entire section.